### PR TITLE
feature: add SmartDocuments global and zaaktype switches

### PR DESCRIPTION
**Pull request by hsiliev**
Add global SmartDocuments enable switch (`SMARTDOCUMENTS_ENABLED` environment variable)
Rename all SmartDocuments environment variables
Add switch for SmartDocuments per zaaktype as part of ZAPS

Solves: PZ-3401, PZ-3403, PZ-3404, PZ-3409

### CODE REVIEW

**Review by infonl-marcel — APPROVED** [review]

File: src/main/resources/schemas/V67__zaakafhandelparameters_create_document_enable_column.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Lifely
+ * SPDX-License-Identifier: EUPL-1.2+
+ */
+
+ALTER TABLE ${schema}.zaakafhandelparameters
+    ADD COLUMN document_maken_ingeschakeld BOOL default false;
```
[the remark this answers is not in the document]
Comment (hsiliev) [reply src/main/resources/schemas/V67__zaakafhandelparameters_create_document_enable_column.sql:7]: Now we support only SmartDocuments, so perhaps we can make this more specific. 
@edgarvonk @bas-info-nl any thoughts on this?

**Comment by infojohn** [comment]
This variable should also be added to the helm/values.yaml under smartDocuments:

**Comment by hsiliev** [comment]
> This variable should also be added to the helm/values.yaml under smartDocuments:

Done

File: src/main/resources/schemas/V67__zaakafhandelparameters_create_document_enable_column.sql (hunk 1 above)
[the remark this answers is not in the document]
Comment (edgarvonk) [reply src/main/resources/schemas/V67__zaakafhandelparameters_create_document_enable_column.sql:7]: I would make it specific for SmartDocuments yes. We have no generic support so no reason to pretend we do.

**Review by edgarvonk — APPROVED** [review]

File: .env.example [hunk 2]
```diff
@@ -74,17 +74,19 @@ MAX_FILE_SIZE_MB=80
 OPA_API_CLIENT_MP_REST_URL=http://localhost:8181
 # Open Notifications API secret key
 OPEN_NOTIFICATIONS_API_SECRET_KEY=openNotificatiesApiSecretKey
+# Enables or disables the support for creating a documenta via SmartDocuments.
+SMARTDOCUMENTS_ENABLED=false
 # SmartDocuments API authentication token. Please contact SmartDocuments to request an API token.
-SD_AUTHENTICATION=XXX
+SMARTDOCUMENTS_AUTHENTICATION=XXX
```
Comment (edgarvonk) [inline .env.example:80]: Maybe too late now but I would not change existing env vars in this already large PR. I would have just add the new `SMARTDOCUMENTS_ENABLED` env var and leave all the others to `SD_xxx` for now and separately discuss how we want to go about renaming those in a separate change.

Comment (edgarvonk) [reply .env.example:80]: For now it's probably ok but for future I think we need to take a lot care in breaking changes in the Helm Charts. It requires a separate process.

**Pull request merged**